src/context/display/findReferences.ml

```diff
@@ -25,18 +26,74 @@ let find_references tctx com with_definition name pos kind =
 		(try loop acc (Hashtbl.find relations p)
 		with Not_found -> acc)
 	) symbols [] in
-	let usages = List.sort (fun p1 p2 ->
-		let c = compare p1.pfile p2.pfile in
-		if c <> 0 then c else compare p1.pmin p2.pmin
-	) usages in
 	t();
-	Display.ReferencePosition.set ("",null_pos,SKOther);
-	DisplayException.raise_positions usages
+	usages
 
-let find_references tctx com with_definition =
+let collect_reference_positions com =
 	let name,pos,kind = Display.ReferencePosition.get () in
-	if pos <> null_pos then find_references tctx com with_definition name pos kind
-	else DisplayException.raise_positions []
+	match kind, com.display.dms_kind with
+	| SKField (cf,Some cl_path), DMUsage (_,find_descendants,find_base) when find_descendants || find_base ->
+		let collect() =
+			let c =
+				let rec loop = function
+					| [] -> raise Exit
+					| TClassDecl c :: _ when c.cl_path = cl_path -> c
+					| _ :: types -> loop types
+				in
+				loop com.types
```

Comment:
Is there a particular reason we store the class' path and then dig it up here, instead of just storing the `tclass` reference itself like we usually do?

Reply:
Yes. 
`tclass` instances are not physically equal [at this moment](https://github.com/RealyUniqueName/haxe/blob/1a5ae13/src/context/display/displayEmitter.ml#L105)  and at the moment of `find_references` invocation.
Maybe that's a bug, but I didn't investigate that.